### ISSUE TITLE
Command+B for Play the project shortcut sometimes takes two presses to activate

### ISSUE THREAD

**Issue by inDigiNeous** [godotengine/godot#35429]
**Godot version:**
3.2.rc2.official

**OS/device including version:**
Macbook Pro 2014, macOS 10.14.16

**Issue description:**
Since updating to 3.2rc1, sometimes, but not always, usually when initially loading a project and trying to play it with Command+B keyboard shortcut, it takes two presses of the shortcut to actually start the project.

I did not see this happen with the latest 3.1 version. Reporting this for now,maybe will look into it more by compiling the engine and seeing what happens, but unfortunately don't have more info about this.

**Steps to reproduce:**
I don't know how to 100% reproduce this, as sometimes the Command+B works perfectly, but usually when initially I have loaded a project, and try to play it with Command+B, it needs two presses to start.

This might have something to do with the mouse focus, although I'm not sure, sometimes moving the mouse to a different location on the UI seems to help, this might be totally bogus though, I don't know.

**Minimal reproduction project:**

This seems to happen on any project I've already made.

**Comment by inDigiNeous** [comment]
![Godot command_b bug](https://user-images.githubusercontent.com/1212726/72870938-35c3d480-3cf2-11ea-8169-cf461277a778.gif)

Here you can see the behaviour. Observe the keyboard shortcuts being displayed on the bottom right of the animated gif.

**Comment by hiulit** [comment]
* Godot Engine v3.2.stable.official
* MacBook Pro (Retina, 15-inch, Mid 2015)
* OpenGL ES 3.0 Renderer: Intel Iris Pro OpenGL Engine

It's happening to me too.

I can't seem to find a pattern. Sometimes it happens and sometimes it doesn't.

**Comment by Calinou** [comment]
Can anyone reproduce this on Windows or Linux (with the <kbd>F5</kbd> shortcut instead)?

@inDigiNeous @hiulit Can you reproduce this if you change the **Editor > Play** shortcut to something like <kbd>F5</kbd> in the Editor Settings?

**Comment by hiulit** [comment]
@Calinou I just changed it to <kbd>Enter</kbd>, and it works every time I press it.
I also tried <kbd>F5</kbd> (which on a Mac laptop it's actually <kbd>fn</kbd> + <kbd>F5</kbd>) and it also works every time.

**Comment by Calinou** [comment]
For macOS users, can you reproduce this on any earlier 3.2 build (from alpha0 to rc4)? You can download them [here](https://downloads.tuxfamily.org/godotengine/3.2/).

**Comment by akien-mga** [comment]
Might be worth checking if there's anything else bound to Cmd+B in the editor shortcuts that might take precedence.

**Comment by hiulit** [comment]
@Calinou @akien-mga I found out that there are 3 shortcuts that use <kbd>cmd</kbd> + <kbd>B</kbd>.

<img width="868" alt="Editor-Play" src="https://user-images.githubusercontent.com/481475/73435053-338ff480-4348-11ea-821e-46d2c78ba87e.png">
<img width="867" alt="Script-Text-Editor-Go-To-Next-Bookmark" src="https://user-images.githubusercontent.com/481475/73435054-338ff480-4348-11ea-8ec5-da055176ea87.png">
<img width="874" alt="Tile-Map-Editor-Select" src="https://user-images.githubusercontent.com/481475/73435055-34288b00-4348-11ea-941c-14651659264b.png">

**Comment by akien-mga** [comment]
Can you test disable the script text editor one and/or the tile map editor one to see if either of those is conflicting?

It could also explain the seemingly "random" behavior of the issue, maybe it's only not working the first time when you have the script editor or a tilemap in focus.

**Comment by inDigiNeous** [comment]
@akien-mga Can confirm that either changing the shortcut to F5 or disabling the other bindings fixes this. Thanks. Didn't think there were conflicting keyboard shortcuts for such a global operation. Probably default behaviour should be such that these do not conflict.

**Comment by inDigiNeous** [comment]
> To macOS users: can you reproduce this on any earlier 3.2 build (from alpha0 to rc4)? You can download them [here](https://downloads.tuxfamily.org/godotengine/3.2/).

I have seen this on all 3.2 RC builds, and now on 3.2 master also. Didn't see this on 3.1.1 or 3.1.2.

**Comment by hiulit** [comment]
> Can you test disable the script text editor one and/or the tile map editor one to see if either of those is conflicting?

 @akien-mga I've tried disabling the script editor shortcut and it seems to work better, but not a 100%.

Also, I'm adding **Play Scene** <kbd>Cmd</kbd> + <kbd>R</kbd> into the mix.

<img width="879" alt="Play-Scene" src="https://user-images.githubusercontent.com/481475/73446194-9e4b2b00-435c-11ea-9486-a476541c6391.png">
<img width="881" alt="Refresh-Graph" src="https://user-images.githubusercontent.com/481475/73446195-9e4b2b00-435c-11ea-9644-25ff1f532998.png">

It behaves as bad as **Play** <kbd>Cmd</kbd> + <kbd>B</kbd>.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by Sakari369** [comment]
> Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

Hey! Still can see in 3.2.4 beta4 ! Feels like sometimes I have to click the main view to make it work. Not sure though, after I moved to using F5 i have not seen it happen, so it has something to do probably with Cmd+B being bound to multiple things or having to handle a key combination differently than a single key ? Just guessing.

**Comment by EricEzaM** [comment]
Might be fixed in master, needs testing.

**Comment by coppolaemilio** [comment]
I was having issues running the game by pressing `command` + `B` on my Mac (M1, Ventura 13.2.1).
To fix it I had to remove a shortcut that was already using the same command as noted on a comment above. 

Godot version: 4.0.1.stable

**Comment by fdstevex** [comment]
Still happening in 4.1.3.  Cmd-B is also bound to Bookmark by default, and when the cursor is sitting in an editor, that seems to take precedence. Removed that binding and Cmd-B to Run Project works reliably.

**Comment by coppolaemilio** [comment]
I guess the solution would be to change the shortcut for bookmark to Cmd-shift-B in mac?

**Comment by fdstevex** [comment]
I went looking for where this is defined and it seems like the current default is Option-B.

So maybe this issue is only hitting people who have a keymap that was created before this change?

https://github.com/godotengine/godot/blob/4c5a1484527ec5f8e320a24d0ee4c4fde0514776/editor/gui/editor_run_bar.cpp#L381C2-L381C2

**Comment by bruvzg** [comment]
> I went looking for where this is defined and it seems like the current default is Option-B.

`KeyModifierMask::META` is Command not Option, Option+B can't be used since it's used to input some special characters (like `∫`).

**Comment by bruvzg** [comment]
> yeah the fix is as you said, change the shortcut for Go to Next Bookmark to Cmd-Shift-B.

It's already used as well as most others stuff (and already have other conflicts as well):
https://github.com/godotengine/godot/blob/4c5a1484527ec5f8e320a24d0ee4c4fde0514776/editor/plugins/script_text_editor.cpp#L2484-L2486
https://github.com/godotengine/godot/blob/4c5a1484527ec5f8e320a24d0ee4c4fde0514776/editor/plugins/script_text_editor.cpp#L2495-L2495
https://github.com/godotengine/godot/blob/4c5a1484527ec5f8e320a24d0ee4c4fde0514776/editor/plugins/canvas_item_editor_plugin.cpp#L5352

It's quire hard to find free combinations (at least without using 3+ modifiers) and avoid conflicts with input and system shortcuts. So it's definitely not the only one with this issue.

**Comment by fdstevex** [comment]
Yeah no kidding.  "Run Project" is probably one of the most important shortcuts and the current behaviour where it works sometimes is maddening.

**Comment by coppolaemilio** [comment]
I would be in favor of removing the bookmark shortcut for mac users just to get the running one by default
